engine/apps/mobile_app/fcm_relay.py

```diff
@@ -49,14 +50,17 @@ def post(self, request):
 
 
 @shared_dedicated_queue_retry_task(
-    autoretry_for=(Exception,), retry_backoff=True, max_retries=1 if settings.DEBUG else 3
+    autoretry_for=(Exception,), retry_backoff=True, max_retries=1 if settings.DEBUG else 5
 )
 def fcm_relay_async(token, data, apns):
     message = Message(token=token, data=data, apns=deserialize_apns(apns))
 
-    task_logger.debug(f"Sending message to FCM: {message}")
-    result = FCMDevice(registration_id=token).send_message(message)
-    task_logger.debug(f"FCM response: {result.success}")
+    # https://firebase.google.com/docs/cloud-messaging/http-server-ref#interpret-downstream
+    response = FCMDevice(registration_id=token).send_message(message)
+    task_logger.debug(f"FCM response: {response}")
+
+    if isinstance(response, FirebaseError):
```

Comment:
Eventually would it be worth to check the status code and retry only if it makes sense?

Reply:
Yes, I think this makes sense not to retry for 4xx errors for example. But I'm not sure how this will behave on production just yet, so I propose to narrow this down later.